jobsdb/unionQuery_test.go

```diff
@@ -111,3 +114,183 @@ func TestMultiTenantHandleT_GetAllJobs(t *testing.T) {
+			CustomValFilters: []string{customVal},
+			JobsLimit:        10,
+		}, 10, nil)
+		require.NoError(b, err)
+	}
+}
+
+func benchmarkConcurrentUnionQuery(b *testing.B, jobsdb MultiTenantHandleT, testWorkspacePickup map[string]int, customVal string, concurrency int) {
+	b.ResetTimer()
+	var wg sync.WaitGroup
+	for i := 0; i < 10; i++ {
+		wg.Add(concurrency)
+		for i = 0; i < concurrency; i++ {
+			go func() {
+				defer wg.Done()
+				_, err := jobsdb.GetAllJobs(context.Background(), testWorkspacePickup, GetQueryParamsT{
+					CustomValFilters: []string{customVal},
+					JobsLimit:        10,
+				}, 10, nil)
+				require.NoError(b, err)
+			}()
+		}
+		wg.Wait()
+	}
+}
+
+func benchmarkParallelQuery(b *testing.B, jobsdb MultiTenantLegacy, testWorkspacePickup map[string]int, customVal string) {
+	b.ResetTimer()
+	for i := 0; i < 10; i++ {
+		concurrencies := len(testWorkspacePickup)
+		var wg sync.WaitGroup
+		wg.Add(concurrencies)
+		for workspaceID, count := range testWorkspacePickup {
+			go func(workspaceID string, count int) {
+				defer wg.Done()
+				_, err := jobsdb.GetAllJobs(context.Background(), map[string]int{workspaceID: count}, GetQueryParamsT{
+					CustomValFilters: []string{customVal},
+					WorkspaceFilter:  []string{workspaceID},
+					JobsLimit:        10,
+				}, 10, nil)
+				require.NoError(b, err)
+			}(workspaceID, count)
+		}
+		wg.Wait()
+	}
+}
+
+func benchmarkConcurrentParallelQuery(b *testing.B, jobsdb MultiTenantLegacy, testWorkspacePickup map[string]int, customVal string, concurrency int) {
+	b.ResetTimer()
+	for i := 0; i < 10; i++ {
+		var wg sync.WaitGroup
+		wg.Add(concurrency)
+		for i = 0; i < concurrency; i++ {
+			go func() {
+				defer wg.Done()
+				parallelism := len(testWorkspacePickup)
+				var wgInternal sync.WaitGroup
+				wgInternal.Add(parallelism)
+				for workspaceID, count := range testWorkspacePickup {
+					go func(workspaceID string, count int) {
+						defer wg.Done()
+						_, err := jobsdb.GetAllJobs(context.Background(), map[string]int{workspaceID: count}, GetQueryParamsT{
+							CustomValFilters: []string{customVal},
+							WorkspaceFilter:  []string{workspaceID},
+							JobsLimit:        10,
+						}, 10, nil)
+						require.NoError(b, err)
+					}(workspaceID, count)
+				}
+				wgInternal.Wait()
+			}()
+		}
+		wg.Wait()
+	}
+}
+
+func generateJobs(size int) ([]*JobT, map[string]int) {
+	workspaces := []string{"testWorkspace1", "testWorkspace2", "testWorkspace3", "testWorkspace4", "testWorkspace5", "testWorkspace6", "testWorkspace7", "testWorkspace8", "testWorkspace9", "testWorkspace10", "testWorkspace11", "testWorkspace12", "testWorkspace13", "testWorkspace14", "testWorkspace15", "testWorkspace16", "testWorkspace17", "testWorkspace18", "testWorkspace19", "testWorkspace20"}
+	jobs := make([]*JobT, size)
+	testWorkspacePickup := map[string]int{}
+	for i := 0; i < size; i++ {
+		workspaceId := workspaces[rand.Int()%len(workspaces)]
```

Comment:
I don't like that we don't control the number of workspaces in the pickup. This could potentially give very different results. If very unlucky we could end up with always the same workspace(s) in the pickup map. Can we make the number of workspaces that we want in the pickup a part of the benchmark configuration?

For example:
1. union query 1k jobs and 5 workspaces
2. union query 1k jobs and 20 workspaces
3. union query 1k jobs and 50 workspaces
4. ...

Then repeat with parallel query then repeat with 10k jobs and again 100k jobs.